File: app/src/main/java/com/plweegie/android/squash/data/RepoDao.java
package com.plweegie.android.squash.data;

import androidx.lifecycle.LiveData;
import androidx.room.Dao;
import androidx.room.Insert;
import androidx.room.OnConflictStrategy;
import androidx.room.Query;

import java.util.List;

@Dao
public interface RepoDao {

    @Query("SELECT * FROM repos")
    LiveData<List<RepoEntry>> getFavorites();

    @Query("SELECT * FROM repos")
    List<RepoEntry> getFavoritesDirectly();

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    void insertFavorite(RepoEntry repo);

    @Query("DELETE FROM repos WHERE repoId = :repoId")
    void deleteSelected(long repoId);

    @Query("DELETE FROM repos")
    void deleteAll();
}

File: app/src/main/java/com/plweegie/android/squash/data/RepoDao.kt
package com.plweegie.android.squash.data

import androidx.lifecycle.LiveData
import androidx.room.Dao
import androidx.room.Insert
import androidx.room.OnConflictStrategy
import androidx.room.Query

@Dao
interface RepoDao {

    @get:Query("SELECT * FROM repos")
    val favorites: LiveData<List<RepoEntry>>

    @Query("SELECT * FROM repos")
    suspend fun getFavoritesAsync(): List<RepoEntry>

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    suspend fun insertFavorite(repo: RepoEntry)

    @Query("DELETE FROM repos WHERE repoId = :repoId")
    suspend fun deleteSelected(repoId: Long)

    @Query("DELETE FROM repos")
    suspend fun deleteAll()
}
